### PR TITLE
Fix `cast_to_rectilinear_grid` floating-point error

### PR DESCRIPTION
**Pull request by GuillaumeBroggi**
The current implementation can introduce a floating-point error when the `spacing` is a float.

For instance, the following would fail:
```python
import pyvista as pv

grid = pv.ImageData(
    dimensions=(10, 10, 10),
    spacing=(27.88888888888889, 28.11111111111111, 28.22222222222222),
    origin=(-126.0, -127.0, -127.0),
)
rectilinear = grid.cast_to_rectilinear_grid()
assert rectilinear.bounds == grid.bounds
```

The proposed PR uses `np.linspace` to generate the rectilinear coordinates so floating-point errors do not accumulate.

### CODE REVIEW

File: tests/core/test_grid.py [hunk 1]
```diff
@@ -937,6 +937,18 @@ def test_cast_uniform_to_rectilinear():
     assert rectilinear.bounds == grid.bounds
 
 
+def test_cast_image_data_with_float_spacing_to_rectilinear():
+    grid = pv.ImageData(
```
Comment (tkoyama010) [inline tests/core/test_grid.py:941]:
```suggestion
    # https://github.com/pyvista/pyvista/pull/6656
    grid = pv.ImageData(
```

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Comment by GuillaumeBroggi** [comment]
As a side note, the proposed PR does not degrade performances.

```python
%%timeit
grid = pv.ImageData(
    dimensions=(10000, 10000, 10000),
    spacing=(27.88888888888889, 28.11111111111111, 28.22222222222222),
    origin=(-126.0, -127.0, -127.0),
)
rectilinear = grid.cast_to_rectilinear_grid()
```
Current implementation:
> `446 μs ± 10.5 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)`
Proposed PR:
> `304 μs ± 2.37 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)`

**Comment by tkoyama010** [comment]
Glad to hear it :)

**Review by MatthewFlamm — APPROVED** [review]
Thanks!  This would be better without the unrelated formatting changes however.  If the formatting needs to be changed in this file it would be better in a separate PR.

**Comment by GuillaumeBroggi** [comment]
@MatthewFlamm Good point; I did not notice the formatting changed; that's a mistake. I'll correct it. Thanks!

**Comment by user27182** [comment]
Looks like the use of single quotes for the formatting may be preferred for the project. See #6657.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
OK. Let's wait for it.

**Comment by GuillaumeBroggi** [comment]
It should be fixed now. I had unintentionally applied black formatter.

**Review by tkoyama010 — APPROVED** [review]

**Review by MatthewFlamm — APPROVED** [review]

**Pull request merged**